src/middlewares/adminRole.js

```diff
@@ -0,0 +1,14 @@
+const userModel = require("../models/userSchema");
+const adminRole = async (req, res, next) => {
+  const { role, userId } = req.user;
+  try {
+    if (role !== "admin") {
```

Comment:
If I'm following the logic here correctly, any user can make a PUT request to update their own role to be an admin, is this correct? This would be insecure since any user could create their own account then assign themselves to be an admin.

In general I'd recommend that only admins can create admin users. It'd be good to add logic to this endpoint that the requesting user must have the admin role otherwise return a 403 forbidden.

But then you may ask, how do we create the first admin user? :eyes:  The first admin user can be created directly in the database. https://www.shellhacks.com/mongodb-create-user-database-admin-root/. You could either proceed with creating any other future admins in the database directly, or expose an endpoint for admin users to be able to create other admin users.